l1-contracts/src/core/libraries/rollup/STFLib.sol

```diff
@@ -95,17 +205,48 @@ library STFLib {
     emit IRollupCore.PrunedPending(proven, pending);
   }
 
+  /**
+   * @notice Calculates the size of the circular storage buffer for temporary block logs
+   * @dev The roundabout size determines how many blocks can be stored in the circular buffer
+   *      before older entries are overwritten. The size is calculated as:
+   *
+   *      roundaboutSize = maxPrunableBlocks() + 1
+   *
+   *      Where maxPrunableBlocks() = epochDuration * (proofSubmissionEpochs + 1)
+   *
+   *      This ensures that:
+   *      - All blocks within the proof submission window remain accessible
+   *      - At least one additional slot is available for the next block
+   *
+   * @return The number of slots in the circular storage buffer
+   */
   function roundaboutSize() internal view returns (uint256) {
     // Must be ensured to contain at least the last provable even after a prune.
     return TimeLib.maxPrunableBlocks() + 1;
   }
 
   /**
-   * A block is stale if it cannot be accessed in the roundabout anymore.
-   * Consider a roundabout with 5 blocks, and the pending block is 7.
-   * Then it would look like [5,6,7,3,4]
-   * So 2 and below are stale. That is, 7 - 5 and below (inclusive)
-   * So _blockNumber + size <= pending is the condition for a block to be stale.
+   * @notice Determines if a block number is stale (no longer accessible in circular storage)
+   * @dev A block is considered stale if it can no longer be accessed in the circular storage buffer.
+   *      The staleness is determined by the relationship between the block number, current pending
+   *      block, and the buffer size.
+   *
+   *      Example with roundabout size 5 and pending block 7:
+   *      Circular buffer state: [block3, block4, block5, block6, block7]
+   *      Buffer indices:        [3,     4,     0,     1,     2    ]
```

Comment:
It might be me with a tiny micro brain, but the buffer indices here makes it more confusing for me to read 😆